File: mobile/src/com/fancytrade/entities/Comment.java
package com.fancytrade.entities;

import com.fancytrade.utils.DateUtils;
import com.fancytrade.utils.Statics;

import java.util.Date;

public class Comment implements Comparable<Comment> {

    private int id;
    private Publication publication;
    private User user;
    private String content;
    private Date date;

    public Comment() {
    }

    public Comment(int id, Publication publication, User user, String content, Date date) {
        this.id = id;
        this.publication = publication;
        this.user = user;
        this.content = content;
        this.date = date;
    }

    public Comment(Publication publication, User user, String content, Date date) {
        this.publication = publication;
        this.user = user;
        this.content = content;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Publication getPublication() {
        return publication;
    }

    public void setPublication(Publication publication) {
        this.publication = publication;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @Override
    public int compareTo(Comment comment) {
        switch (Statics.compareVar) {
            case "Publication":
                return this.getPublication().getContent().compareTo(comment.getPublication().getContent());
            case "User":
                return this.getUser().getEmail().compareTo(comment.getUser().getEmail());
            case "Content":
                return this.getContent().compareTo(comment.getContent());
            case "Date":
                DateUtils.compareDates(this.getDate(), comment.getDate());

            default:
                return 0;
        }
    }

}

File: symfony/src/Entity/Comment.php
<?php

namespace App\Entity;

use App\Repository\CommentRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CommentRepository::class)]
class Comment implements \JsonSerializable
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"content is required")]
    private ?string $content = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message:"date is required")]
    private ?\DateTimeInterface $date = null;

    #[ORM\ManyToOne(inversedBy: 'Comment')]
    private ?Publication $publication = null;

    #[ORM\ManyToOne]
#[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id_user')]
    private ?User $user = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getContent(): ?string
    {
        return $this->content;
    }

    public function setContent(string $content): self
    {
        $this->content = $content;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getPublication(): ?Publication
    {
        return $this->publication;
    }

    public function setPublication(?Publication $publication): self
    {
        $this->publication = $publication;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function jsonSerialize(): array
    {
        return array(
            'id' => $this->id,
            'publication' => $this->publication,
            'user' => $this->user,
            'content' => $this->content,
            'date' => $this->date->format("d-m-Y")

        );
    }

    public function constructor($publication, $user, $content, $date)
    {
        $this->publication = $publication;
        $this->user = $user;
        $this->content = $content;
        $this->date = $date;

    }
}
